src/index.js

```diff
@@ -196,12 +196,20 @@ class HCaptcha extends React.Component {
       if (onError) onError(event);
     }
 
-    execute () {
+    async execute (opts) {
       const { isApiReady, isRemoved, captchaId } = this.state;
 
-      if (!isApiReady || isRemoved) return
+      if (!isApiReady || isRemoved) return;
+
+      if (opts && opts.async === true) {
+        return hcaptcha.execute(captchaId, { async: true });
+      }
+
+      // else if (opts && typeof opts.async !== "boolean") {
+      //    TODO: catch bad opts and throw custom hcaptcha error
+      // }
 
-      hcaptcha.execute(captchaId)
+      hcaptcha.execute(captchaId);
```

Comment:
Since you are not using `await` here you don't need `async` in front of the method. We also have a few parameters that can be passed in via the options object (intended for enterprise). So let's make a simple check for typeof equal to object, else set it to null and let the hCaptcha API handle the rest.

```js
execute (opts=null) {
      const { isApiReady, isRemoved, captchaId } = this.state;

      if (!isApiReady || isRemoved) return;
      
      if (opts && typeof opts !== "object") {
        opts = null;
      }

      return hcaptcha.execute(captchaId, opts);
}
```

Reply:
Done